src/components/renderer/field/form-field-renderer.component.tsx

```diff
@@ -148,25 +148,41 @@ export const FormFieldRenderer = ({ field, valueAdapter, repeatOptions }: FormFi
       />
     );
   }
+
   return (
     <ErrorBoundary FallbackComponent={ErrorFallback} onReset={noop}>
       <Controller
         control={control}
         name={field.id}
         render={({ field: { value, onChange, onBlur } }) => (
           <div>
-            <InputComponent
-              key={`${field.id}-input-component`}
-              field={field}
-              value={value}
-              errors={errors}
-              warnings={warnings}
-              setFieldValue={(val) => {
-                onChange(val);
-                onAfterChange(val);
-                onBlur();
-              }}
-            />
+            {field.type === 'programState' && historicalValue?.value ? (
```

Comment:
I'm afraid of the coupling this potentially introduces. I also don't understand why we are inferring the current value from the "historicalValue". Do you know why `FormFieldValueAdapter.getInitialValue(...)` doesn't suffice? What is the problem we are dealing with here?

Reply:
The value of `FormFieldValueAdapter.getInitialValue(...)` is undefined, thats why i used this approach. The intention is to pre-populate the field as opposed to having the user click the `usePreviousValue` button. @slubwama can provide more context on the issue

Reply:
> The value of FormFieldValueAdapter.getInitialValue(...) is undefined, thats why i used this approach.

Shouldn't be investigating why the value is undefined? My concern is ensuring that we consistently manage value binding.